From 65cab19a5c817fc2baa420c0f745fc534dbc281d Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Mon, 18 Apr 2022 09:34:50 -0300
Subject: [PATCH] Disable overlay-scrolling

gtk-overlay-scrolling property is ignored, only way to enable is GTK_OVERLAY_SCROLLING=1
---
 gtk/gtkscrolledwindow.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkscrolledwindow.c b/gtk/gtkscrolledwindow.c
index e901389..9cfbc3d 100644
--- a/gtk/gtkscrolledwindow.c
+++ b/gtk/gtkscrolledwindow.c
@@ -725,7 +725,7 @@ gtk_scrolled_window_class_init (GtkScrolledWindowClass *class)
    */
   properties[PROP_OVERLAY_SCROLLING] =
       g_param_spec_boolean ("overlay-scrolling", NULL, NULL,
-                            TRUE,
+                            FALSE,
                             G_PARAM_READWRITE | G_PARAM_STATIC_NAME | G_PARAM_EXPLICIT_NOTIFY);
 
   /**
@@ -3922,7 +3922,10 @@ gtk_scrolled_window_update_use_indicators (GtkScrolledWindow *scrolled_window)
 
   g_object_get (settings, "gtk-overlay-scrolling", &overlay_scrolling, NULL);
 
-  use_indicators = overlay_scrolling && priv->overlay_scrolling;
+  use_indicators = FALSE;
+
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 1)
+    use_indicators = overlay_scrolling && priv->overlay_scrolling;
 
   if (priv->use_indicators != use_indicators)
     {
-- 
2.55.0

